### DISCUSSION TITLE
Can I use daphne as a drop in replacement for gunicorn in production?

### DISCUSSION THREAD

**Discussion by skfidan-apothecare** [django/daphne#541] [Q&A] [answered]
Generally people recommend running Daphne and Gunicorn while using nginx to route requests to the appropriate application.

I would set this up but it would mean moving from our current managed hosting with automated build and deploy. We are a small team so we would like to keep the benefits of managed hosting over setting up our own Ubuntu server.

I know Daphne can be used for HTTP and Websocket requests but should it be in a live production environment? Are there likely to be any issues with performance, maintenance, etc?

**Comment by skfidan-apothecare** [comment]
Sorry, for context this is for a Django application

**Comment by bigfootjon** [comment, marked as answer]
It's the reference implementation for ASGI, so it is going to be correct but not necessarily the most feature-rich, advanced, or efficient.

I think for a small team that wants to avoid self-hosting web services using daphne makes sense until you hit scaling limits (daphne is a pure python implementation so it is not going to be fast) or feature limitations (I'm not really aware of any, but they may exist)
